### ISSUE TITLE
/waypoint-entrypoint: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found

### ISSUE THREAD

**Issue by gtirloni** [hashicorp/waypoint#572]
Using a `waypoint` binary compiled from the repository returns this error (everything works fine using the released `0.1.2` binary). Initially reported [here](https://github.com/hashicorp/waypoint/issues/530#issuecomment-709965248) by @rvbuelow.

**Environment**

Fedora Linux 33
Go 1.15.2 amd64
go-bindata 3.1.0
Waypoint 6ebc4546 (6ebc4546)

**Build Steps**

```
cd waypoint
make bin
```

Output:

```
GOOS=linux GOARCH=amd64 go build -o ./internal/assets/ceb/ceb ./cmd/waypoint-entrypoint
cd internal/assets && go-bindata -pkg assets -o prod.go -tags assetsembedded ./ceb
CGO_ENABLED=0 go build -ldflags "-X "github.com/hashicorp/waypoint/internal/version".GitCommit=$(git rev-parse --short HEAD)$(test -n "`git status --porcelain`" && echo "+CHANGES" || true) -X "github.com/hashicorp/waypoint/internal/version".GitDescribe=$(git describe --tags --always --match "v*")" -tags assetsembedded -o ./waypoint ./cmd/waypoint
go build -tags assetsembedded -o ./waypoint-entrypoint ./cmd/waypoint-entrypoint
```

**Steps to Reproduce Error**

```
(main) ~/code/hashicorp/waypoint-examples/docker/nodejs $ waypoint init
✓ Configuration file appears valid
✓ Connection to Waypoint server was successful
✓ Project "example-nodejs" and all apps are registered with the server.
✓ Plugins loaded and configured successfully
✓ Authentication requirements appear satisfied.

Project initialized!

You may now call 'waypoint up' to deploy your project or
commands such as 'waypoint build' to perform steps individually.
```

```
(main) ~/code/hashicorp/waypoint-examples/docker/nodejs $ waypoint up
» Building...
Creating new buildpack-based image using builder: heroku/buildpacks:18
✓ Creating pack client
✓ Building image
 │ [exporter] Reusing 1/1 app layer(s)
 │ [exporter] Reusing layer 'launcher'
 │ [exporter] Reusing layer 'config'
 │ [exporter] Adding label 'io.buildpacks.lifecycle.metadata'
 │ [exporter] Adding label 'io.buildpacks.build.metadata'
 │ [exporter] Adding label 'io.buildpacks.project.metadata'
 │ [exporter] *** Images (5b7fb0507eef):
 │ [exporter]       index.docker.io/library/example-nodejs:latest
 │ [exporter] Adding cache layer 'heroku/nodejs-engine:nodejs'
 │ [exporter] Adding cache layer 'heroku/nodejs-engine:toolbox'
✓ Injecting entrypoint binary to image

» Deploying...
✓ Setting up waypoint network
✓ Starting container

» Releasing...

The deploy was successful! A Waypoint deployment URL is shown below. This
can be used internally to check your deployment and is not meant for external
traffic. You can manage this hostname using "waypoint hostname."

           URL: https://steadily-polished-moose.waypoint.run
Deployment URL: https://steadily-polished-moose--v1.waypoint.run
```

Here are the containers:

```
$ docker ps -a
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                     PORTS                              NAMES
cde50d4e3f0f        example-nodejs:latest       "/waypoint-entrypoin…"   2 minutes ago       Exited (1) 2 minutes ago                                      example-nodejs-01EMSTF97JZ17T0YX2MMPGZQX5
953cdf04bb52        hashicorp/waypoint:latest   "/usr/bin/waypoint s…"   4 minutes ago       Up 4 minutes               0.0.0.0:9701-9702->9701-9702/tcp   waypoint-server
```

And the logs for the stopped container:

```
$ docker logs cde50d4e3f0f
/waypoint-entrypoint: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /waypoint-entrypoint)
```

The /waypoint-entrypoint:

```
$ docker run --rm -ti --entrypoint=/bin/sh example-nodejs:latest
$ ldd /waypoint-entrypoint
/waypoint-entrypoint: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /waypoint-entrypoint)
	linux-vdso.so.1 (0x00007ffd9d3d0000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f3cd5a57000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3cd5666000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f3cd5c76000)
```

**Comment by mitchellh** [comment]
Hm, I didn't think think there would be any dynamic linking necessary for the entrypoint.

When I run this locally its clean:

```sh
root@34d7d7e401d1:/# ldd /waypoint-entrypoint
	not a dynamic executable
```

Is this perhaps because you're on Linux yourself? Can you try putting a `CGO_ENABLED=0` in front of `GOOS=linux GOARCH=amd64 go build -o ./internal/assets/ceb/ceb ./cmd/waypoint-entrypoint`? Maybe cgo is causing it.

**Comment by evanphx** [comment]
Yes, this is a mismatch between the libc env of your linux system where you compiled the entrypoint and the one inside the created nodejs container. This is a bug in our `make bin` that we're not building with `CGO_ENABLED=0`.

**Comment by mitchellh** [comment]
Fix by #586 will be part of the next WP release.
